src/webui/www/private/scripts/client.js

```diff
@@ -579,18 +582,30 @@ window.addEvent('load', function() {
                         // re-select previously selected rows
                         torrentsTable.reselectRows(torrentsTableSelectedRows);
                 }
-                clearTimeout(syncMainDataTimer);
-                syncMainDataTimer = syncMainData.delay(getSyncMainDataInterval());
+                requestInProgress = false;
+                syncData(getSyncMainDataInterval())
+                $('syncing_div').set('text', '');
             }
-        }).send();
+        });
+        requestInProgress = true;
+        request.send();
     };
 
     updateMainData = function() {
         torrentsTable.updateTable();
-        clearTimeout(syncMainDataTimer);
-        syncMainDataTimer = syncMainData.delay(100);
+        syncData(100);
     };
 
+    const syncData = function(delay) {
+        if (!requestInProgress){
+            clearTimeout(syncMainDataTimer);
+            syncMainDataTimer = syncMainData.delay(delay);
+        }
+        else {
+            $('syncing_div').set('text', 'QBT_TR(Syncing...)QBT_TR[CONTEXT=HttpServer]');
```

Comment:
I've done some testing and I couldn't trigger this message to show.
So is this message really useful to user? Also I *suspect* there is a possibility that the message will appear and disappear quickly and it becomes a distraction for users.

Reply:
It will only really be noticeable if the the request takes long enough. "Long enough" being at least 4 or 5 seconds, which happens if the machine is slow relative to the number of torrents added. This can be emulated also by having a connection slower than GPRS speeds between the WebUi and qbit.

In my case, I have a slow machine with many hundreds of torrents. My connection is more than fast enough, but the API sync takes like 4 to 5 seconds, so the syncing message appears for that length of time. I suppose that it might just flash briefly if you are not quite stressing the machine that much. And of course, if your machine is really fast and/or you have very few torrents, the message will be almost impossible to trigger even if you are spamming category changes.

I still think it is useful for feedback, and not really distracting, since it appears on the lower right corner.

Reply:
> So is this message really useful to user?

I have to agree with this. I'm not sure what the user gains by displaying this message, especially because qBittorrent already displays a message if it's unable to sync.

Reply:
> qBittorrent already displays a message if it's unable to sync.

that message only appears upon failure, which is different. This one indicates that the data will be updated in the future, as a result of a request sent after some user action. For instance, suppose a user changes categories while a sync request is in progress. Then, though the new category appears selected, the torrent table won't be updated immediately, only after the sync response arrives. The `Syncing...` message lets the user know that everything is processing fine/not frozen and that they just have to wait a bit.

Reply:
Like I said, the message is primarily useful for users with very slow connections or with very heavily-loaded instances, where the response to each sync request takes a long time to arrive. 

For everyone else, I don't think it is a nuisance. However, this statement is purely a matter of opinion, so if you strongly prefer that it doesn't show, I'll take it out.

Reply:
@Chocobo1 @Piccirello  So what will it be? Should the message show or should it go? I'd like to get this PR merged either way.

Reply:
@Chocobo1 @Piccirello 
While I waited for your replies I thought I could try using a spinner instead of the text. However, I cannot seem to put it on the right-hand corner.

Here is what I am using instead of the `syncing_div` inside `index.html`:

```html
<div id="syncing_spinner" class="mochaSpinner"></div>
```

and here is my naive attempt at making it appear on the bottom-right corner of the window instead of the bottom-left corner, which did not work, inside `style.css`:

```css
#syncing_spinner {
    float: right;
}
```

Reply:
I can't really help with html.. however I like the idea of using a spinner.
In my imagination, when ever a sync happens the spinner will appear and start spinning and it will disappear when the sync finished. Or we can always display the spinner and let it be in either spin/stop state.

Reply:
@Chocobo1 

>I can't really help with html.. however I like the idea of using a spinner.

Yeah my main problem with having it on the bottom left corner is that it is frequently obstructed by the URL hover tooltip in Firefox (I think this is also the case for the other major browsers).

> In my imagination, when ever a sync happens the spinner will appear and start spinning and it will disappear when the sync finished.

Right now, the spinner/text appears whenever the user does stuff that changes the view while there is an outstanding sync request in progress. The main purpose of this PR is just to prevent the WebUI from spamming the API endpoint to the point of denial-of-service. The spinner just indicates "hey, ideally a sync request would happen at this point (e.g due to selecting a different category) but it won't because another one is already in progress".

I would also prefer to have the spinner appear at _every_ sync, since it gives more and better feedback and transparency of what the WebUI is doing behind the scenes, but I have some reservations. For instance, you said above you were concerned about distractions, and I think it would could be distracting to have the spinner appearing at _every_ sync. It would be like this: sync request sent -> spinner starts -> spinner lasts for however long response takes to arrive -> spinner disappears for 2 seconds (current timeout) -> sync request sent -> spinner starts -> etc.

Again, for me this is not a problem, but let me know what you decide.

> Or we can always display the spinner and let it be in either spin/stop state.

I disagree with always showing it because that would be inconsistent with the way the spinner works in all other windows (e.g. options, add torrent, etc). That could be changed globally, but in another PR.

Reply:
Meanwhile I figured out a way to put the spinner on the right, though it involves a bit of code duplication that is probably not ideal. 

I basically copy-pasted the existing `.mochaSpinner` class (in `Window.css`) but specified and absolute layout constraint on the right margin instead of the left one:

```css
.mochaSpinnerRight {
    display: none;
    position: absolute;
    bottom: 7px;
    right: 6px;
    width: 16px;
    height: 16px;
    background: url(../images/skin/spinner.gif) no-repeat;
}
```

and used that as the div's class in `index.html`:

```html
<div id="syncing_spinner" class="mochaSpinnerRight"></div>
```

@Piccirello better ideas appreciated, my experience with CSS is limited. Anyway, the most important stuff for this PR that has to be decided on is in the previous comment.

Reply:
Seems this UI design could have more discussion, maybe you want to move it (all UI related changes) to another PR? So we can approve & merge "Prevent excessive requests" feature quickly.

Reply:
Ok, I removed all UI stuff. It will be addressed in a future PR.